oak-store-spi/src/test/java/org/apache/jackrabbit/oak/json/JsonSerializerTest.java

```diff
@@ -63,13 +126,59 @@ public void childOrder() throws Exception{
             }
 
         } while (reader.matches(','));
+        Assert.assertTrue(customLogs.getLogs().stream().filter(n -> n.contains("Json is not getting flushed in chunks:")).findFirst().isPresent());
+        assertEquals(expectedOrder, childNames);
+    }
+
+    @Test
+    public void jsonFlushedInChunks() throws Exception {
+        System.setProperty(JsonSerializer.WRITER_FLUSH_THRESHOLD_KEY, "" + 10);
+        NodeBuilder builder = EMPTY_NODE.builder();
+        builder.child("a");
+        builder.child("b");
+        builder.child("c");
+        builder.child("d");
+
+        List<String> expectedOrder = Arrays.asList("a", "c", "b", "d");
+        builder.setProperty(":childOrder", expectedOrder, Type.NAMES);
+
+        NodeState state = builder.getNodeState();
+        String json = serializeJsonFlushedInChunks(state);
 
+        JsopReader reader = new JsopTokenizer(json);
+        List<String> childNames = Lists.newArrayList();
+        reader.read('{');
+        do {
+            String key = reader.readString();
+            reader.read(':');
+            if (reader.matches('{')) {
+                childNames.add(key);
+                reader.matches('}');
+            }
+
+        } while (reader.matches(','));
+        Assert.assertFalse(customLogs.getLogs().stream().filter(n -> n.contains("Json is not getting flushed in chunks:")).findFirst().isPresent());
         assertEquals(expectedOrder, childNames);
     }
 
-    private String serialize(NodeState nodeState){
+    private String serialize(NodeState nodeState) {
         JsopBuilder json = new JsopBuilder();
         new JsonSerializer(json, "{\"properties\":[\"*\", \"-:*\"]}", new BlobSerializer()).serialize(nodeState);
         return json.toString();
     }
+
+    private String serializeJsonFlushedInChunks(NodeState nodeState) throws IOException {
+        PrintWriter pw;
+        OutputStream writerStream;
+        File outFile = temporaryFolder.newFile();
+        OutputStream os = new BufferedOutputStream(FileUtils.openOutputStream(outFile));
+        writerStream = os;
+        pw = new PrintWriter(writerStream);
```

Comment:
the `os` variable is redundant. Make it more concise
```suggestion
        File outFile = temporaryFolder.newFile();
        OutputStream writerStream = new BufferedOutputStream(FileUtils.openOutputStream(outFile));
        PrintWriter pw = new PrintWriter(writerStream);
```